mac/ZRTOPTBL.CPY

```diff
@@ -0,0 +1,316 @@
+*
+* Error routines
+*
+***********************************************************************
+ERROR01  DS    0H
+         WTO   'Error: Open failed',ROUTCDE=11,DESC=(6)
+         MVI   RETCODE,X'08'
+         B     EXIT
+
+ERROR02  DS    0H
+         WTO   'Error: Close failed',ROUTCDE=11,DESC=(6)
+         MVI   RETCODE,8
+         B     EXIT
+
+ERROR03  DS    0H
+         WTO   'Error reading z390 listing',ROUTCDE=11,DESC=(6)
+         MVI   RETCODE,8
+         B     CLOSEFILES
+
+ERROR04  DS    0H
+         WTO   'Error reading HLASM listing',ROUTCDE=11,DESC=(6)
+         MVI   RETCODE,8
+         B     CLOSEFILES
+
+ERROR05  DS    0H
+         WTO   'Error: Internal tables are too small',                 *
+               ROUTCDE=11,DESC=(6)
+         MVI   RETCODE,8
+         B     CLOSEFILES
+
+ERROR06  DS    0H
+         MVC   *+30(8),0(R4)            * insert mnemonic into msg
+         WTO   'Error: mnemonic xxxxxxxx was not expected',            *
+               ROUTCDE=11,DESC=(6)
+         MVI   RETCODE,8
+         LA    R4,&ENTRYLEN.(,R4)       * Bump z390 entry pointer only
+         B     COMPARE_NEXT
+
+ERROR07  DS    0H
+         MVC   *+30(8),0(R6)            * insert mnemonic into msg
+         WTO   'Error: mnemonic xxxxxxxx was expected but not found',  *
+               ROUTCDE=11,DESC=(6)
+         MVI   RETCODE,8
+         LA    R6,&ENTRYLEN.(,R6)       * Bump hlasm entry pointer only
+         B     COMPARE_NEXT
+
+ERROR08  DS    0H
+         MVC   *+30(8),0(R4)            * insert mnemonic into msg
+         WTO   'Error: mnemonic xxxxxxxx incorrectly defined',         *
+               ROUTCDE=11,DESC=(6)
+         MVI   RETCODE,8
+         LA    R4,&ENTRYLEN.(,R4)       * Bump z390 entry pointer
+         LA    R6,&ENTRYLEN.(,R6)       * Bump hlasm entry pointer
+         B     COMPARE_NEXT
+
+***********************************************************************
+*
+* Data areas
+*
+***********************************************************************
+         LTORG ,
+
+SAVEAREA DS    0D
+         DS    18F                     * Savearea for chaining
+GETZ390_RET    DS F                    * Return address
+GETHLASM_RET   DS F                    * Return address
+COMPARE_RET    DS F                    * Return address
+
+Z390PRN  DCB   DSORG=PS,MACRF=GM,DDNAME=Z390PRN,EODAD=Z390END,         *
+               RECFM=FT,LRECL=&Z390LEN
+HLASMPRN DCB   DSORG=PS,MACRF=GM,DDNAME=HLASMPRN,EODAD=HLASMEND,       *
+               RECFM=FT,LRECL=&HLASMLEN
+
+LASTOPCD DS    CL8                     * Last mnemonic processed
+Z390MPT  DS    AL4                     * Ptr to empty entry (=end)
+HLASMMPT DS    AL4                     * Ptr to empty entry (=end)
+
+Z390EOF  DC    XL1'00'
+HLASMEOF DC    XL1'00'
+RETCODE  DC    XL1'00'                 * Return code from program
+Z390REC  DS    CL(&Z390LEN)
+HLASMREC DS    CL(&HLASMLEN)
+
+         DS    0D
+Z390TBL  DS    5000CL(&ENTRYLEN)       * room for 5000 opcodes
```

Comment:
Suggestion only. Consider using an equated value in place of the 5000 here and in the HLASMTBL.declaration.